subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -591,15 +591,19 @@ u16_t lwm2m_get_rd_data(u8_t *client_data, u16_t size)
 			continue;
 		}
 
-		len = snprintf(temp, sizeof(temp), "%s</%u>",
-			       (pos > 0) ? "," : "", obj->obj_id);
-		if (pos + len >= size) {
-			/* full buffer -- exit loop */
-			break;
-		}
+		/* Only report <OBJ_ID> when no instance available */
+		if (obj->instance_count == 0) {
```

Comment:
Hi @rbtchc I don't think this is correct.  Per the spec Section 5.3.1:
"The list of Objects supported and Object Instances available on the LwM2M Client (Security Object ID:0
MUST not be part of this list)." and then it goes on to give a specific example:
"If Objects Instances are already available on the LwM2M Client at the time of registration, then the format would be (for the example client of Appendix F):
</1/0>,</1/1>,</2/0>,</2/1>,</2/2>,</2/3>,</2/4>,</3/0>,</4/0>,</5>"